Uni_life_tracker/app/src/main/java/com/generic/ult/SignUpActivity.java

```diff
@@ -32,11 +35,24 @@ protected void onCreate(Bundle savedInstanceState) {
         textInputEmail = findViewById(R.id.text_input_email);
         textInputPassword = findViewById(R.id.text_password);
 
-        Button startBtn = findViewById(R.id.btn_sign_up);
-        startBtn.setOnClickListener(v -> openMain());
+        Button signupButton = findViewById(R.id.btn_sign_up);
+        signupButton.setOnClickListener(v -> {
+
+            // If all signup credentials are correct, store the credentials
+            // and go into the main page
+            if (validate(textInputName) && validate(textInputUni) &&
```

Comment:
I was wondering if the validate methods need to be here or inside signupInput on line 110 after the user submits

Reply:
What I'm trying to say is that the validate methods helped prevent a wrong signup with invalid credentials

Reply:
Oh I see. I think that might have to be in signupInput because the runs when the button is clicked